extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java

```diff
@@ -150,6 +153,19 @@ public void add(ClassCreator classCreator, FieldDescriptor entityClassFieldDescr
             try (MethodCreator methodCreator = classCreator.getMethodCreator(method.name(), methodReturnTypeDotName.toString(),
                     methodParameterTypesStr)) {
 
+                Set<String> usedNamedParameters = extractNamedParameters(queryString);
+                if (!usedNamedParameters.isEmpty()) {
+                    Set<String> missingParameters = new LinkedHashSet<>(usedNamedParameters);
+                    missingParameters.removeAll(namedParameterToIndex.keySet());
+                    if (!missingParameters.isEmpty()) {
+                        // better error message than "Could not locate ordinal parameter [1], expecting one of []"
```

Comment:
I don't think we need the comment TBH - that comment is best left on the issue (as you have already done)

Reply:
Yes, you're right. I removed it.